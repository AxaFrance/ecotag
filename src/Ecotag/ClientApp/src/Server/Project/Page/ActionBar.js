import React from 'react';
import {useHistory} from 'react-router-dom';
import Button from '@axa-fr/react-toolkit-button';
import './Page.scss';
import ExportButton from "./ExportButton";
import useProjectTranslation from "../../../translations/useProjectTranslation";

export const ActionBar = ({projectId, projectName, isAnnotationClosed, onExport, user}) => {
    const history = useHistory();
    const {translate} = useProjectTranslation();

    const startTaggingButton = () => {
        const path = `/projects/${projectId}/annotations/start`;
        history.push(path);
    };

    if (!projectId) {
        return null;
    }

    return (
        <div className="ft-actionBar">
            {(isAnnotationClosed) ? null : <Button onClick={startTaggingButton} id="startTagging" name={translate('project.project_page.action_bar.start_button_label')}>
                <span className="af-btn-text">{translate('project.project_page.action_bar.start_button_label')}</span>
            </Button>}
            <ExportButton user={user} projectId={projectId} projectName={projectName} onExport={onExport}/>
        </div>
    );
};

export default ActionBar;
