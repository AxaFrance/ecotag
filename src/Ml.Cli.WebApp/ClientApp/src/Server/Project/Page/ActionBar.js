import React from 'react';
import { useHistory } from 'react-router-dom';

import Button from '@axa-fr/react-toolkit-button';
import './Page.scss';

const showEditButton = isAdmin => {
  return isAdmin ? (
    <div>
      <Button className="btn af-btn af-btn--success af-btn--hasiconRight" tabIndex="-1">
        <span className="af-btn__text">Edit Project</span>
        <i className="glyphicon glyphicon-pencil"/>
      </Button>
    </div>
  ) : null;
};

export const ActionBar = ({ currentUser, project, dataset }) => {
  const history = useHistory();

  const startTaggingButton = () => {
    const path = `/projects/${project.id}/annotations/${dataset.id}`;
    history.push(path);
  };

  return (
    <div className="ft-actionBar">
      {showEditButton(currentUser.role === 'ADMIN')}
      <div>
        <Button onClick={() => startTaggingButton()} id="startTagging" name="Start Tagging">
          <span className="af-btn-text">Start Tagging</span>
        </Button>
      </div>
    </div>
  );
};

export default ActionBar;
