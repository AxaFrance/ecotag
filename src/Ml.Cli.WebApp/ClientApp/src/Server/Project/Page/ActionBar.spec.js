import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import {ActionBar} from './ActionBar';
import {BrowserRouter as Router} from "react-router-dom";

jest.mock('react-router-dom', () => {
    // Require the original module to not be mocked...
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        // add your noops here
        useParams: jest.fn(),
        useHistory: jest.fn(),
    };
});

describe('ActionBar', () => {
    it('ActionBar render correctly', async () => {
        const currentUser = {
            role: 'ADMIN'
        };
        const project = {
            "id": "0001",
            "name": "Relev√© d'information",
            "dataSetId": "0004",
            "classification": "Publique",
            "numberTagToDo": 10,
            "createDate": "04/04/2011",
            "typeAnnotation": "NER",
            "text": "Enim ad ex voluptate culpa non cillum eu mollit nulla ex pariatur duis. Commodo officia deserunt elit sint officia consequat elit laboris tempor qui est ex. Laborum magna id deserunt ut fugiat aute nulla in Lorem pariatur. Nostrud elit consectetur exercitation exercitation incididunt consequat occaecat velit voluptate nostrud sunt. Consectetur velit eu amet minim quis sunt in.",
            "labels": [{"name": "Recto", "color": "#212121", "id": 0}, {"name": "Verso", "color": "#ffbb00", "id": 1}, {"name": "Signature", "color": "#f20713", "id": 2}],
            "users": [
                {"annotationCounter": 10,
                    "annotationToBeVerified": 1,
                    "email": "clement.trofleau.lbc@axa.fr"},
                {"annotationCounter": 24,
                    "annotationToBeVerified": 5,
                    "email": "Guillaume.chervet@axa.fr"},
                {"annotationCounter": 35,
                    "annotationToBeVerified": 15,
                    "email": "Gille.Cruchont@axa.fr"}
            ]
        };
        const dataset = {
            "id": "0001",
            "name": "Carte verte",
            "type": "Image",
            "classification": "Publique",
            "numberFiles": 300,
            "createDate": "30/10/2019"
        };
        const { asFragment, getByText } = render(<Router><ActionBar currentUser={currentUser}  dataset={dataset} project={project}/></Router>);
        const messageEl = await waitForElement(() => getByText('Start Tagging'));
        expect(messageEl).toHaveTextContent(
            'Start Tagging'
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

