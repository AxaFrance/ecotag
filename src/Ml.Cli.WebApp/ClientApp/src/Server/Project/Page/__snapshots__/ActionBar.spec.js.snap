// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar ActionBar render correctly 1`] = `
<DocumentFragment>
  <div
    class="ft-actionBar"
  >
    <div>
      <button
        class="btn af-btn af-btn--success af-btn--hasiconRight"
        tabindex="-1"
      >
        <span
          class="af-btn__text"
        >
          Edit Project
        </span>
        <i
          class="glyphicon glyphicon-pencil"
        />
      </button>
    </div>
    <div>
      <button
        class="btn af-btn"
        id="startTagging"
        name="Start Tagging"
      >
        <span
          class="af-btn-text"
        >
          Start Tagging
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
